### PR TITLE
Change how names appear in Devise mailer views

### LINKED ISSUE #803: Give site emails some TLC
**Describe the bug**
The emails from the site could do with some branding & making sure markup and data templating is correct

e.g. invitation to admin account email
image

**Consider**
- Adding logo
- Make sure sender email is correct
- Review copy
- Review use of data (like name) in email
- Think about legal/ privacy / footer information

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Closes #803 

If the user has any first name or last name fields set, use those, otherwise default to email (which must be present).

The invitation email (HTML version) has a PlaceCal logo image (position TBC).

Things to consider:
- email copy (for all emails)
- the sender email appears to be `support@placecal.org`
- Review use of data (like name) in email
- Think about legal/ privacy / footer information

Preview of what it looks like now:

Screenshot_2023-10-26_15-29-36

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -45,6 +45,13 @@ def full_name
     [first_name, last_name].compact_blank.join(' ')
   end
 
+  # used by the (devise) mailers
+  def name_for_email
```
Comment (aaaaargZombies) [inline app/models/user.rb:49]: is this intended to be `name_or_email` ?

Looking at Katjas suggestion in the issue I think maybe this should return an empty string but would need to sort spacing with the `Hello` and the `,`

> Hello Dave Daverson,

vs

> Hello,

**Review by aaaaargZombies — APPROVED** [review]
I think this is good enough, I would like to see the name section changed but it's not blocking for me.

The rest of the email is a huge improvement and looks a lot more professional

File: config/environments/development.rb [hunk 2]
```diff
@@ -44,6 +44,13 @@
 
   config.action_mailer.default_url_options = routes.default_url_options
 
+  asset_url = routes.default_url_options[:protocol]
+  asset_url += "://#{routes.default_url_options[:host]}"
+  asset_url += ":#{routes.default_url_options[:port]}" if routes.default_url_options[:port].present?
+
+  Rails.logger.debug { "action_mailer.asset_host=#{asset_url}" }
```
Comment (aaaaargZombies) [inline config/environments/development.rb:51]: oops look like this is crashing `debugger`?

File: config/environments/production.rb [hunk 3]
```diff
@@ -129,4 +129,11 @@
     authentication: :login,
     enable_starttls_auto: true
   }
+
+  asset_url = routes.default_url_options[:protocol]
+  asset_url += "://#{routes.default_url_options[:host]}"
+  asset_url += ":#{routes.default_url_options[:port]}" if routes.default_url_options[:port].present?
+
+  Rails.logger.debug { "action_mailer.asset_host=#{asset_url}" }
```
Comment (aaaaargZombies) [inline config/environments/production.rb:137]: oops look like this is crashing `debugger`?

**Review by aaaaargZombies — CHANGES_REQUESTED** [review]
sorry spoke too soon, looked at the tests and they are failing, think I commented where the problem is.

**Comment by kimadactyl** [comment]
If we are gonna include the logo lets put it at the top and make it a lot smaller?

GFSC email for ref

Screenshot 2023-10-30 at 16 00 38

**Comment by aaaaargZombies** [comment]
after creating a new user I get redirected but hit this error

`undefined method `greeting_text' for #<ActionView::Base:0x00000000064668>`

coming from `app/views/devise/mailer/invitation_instructions.html.erb`

will the method / class need importing, it appears in the following views

```
app/views/devise/mailer/invitation_instructions.text.erb
app/views/devise/mailer/email_changed.html.erb
app/views/devise/mailer/unlock_instructions.html.erb
app/views/devise/mailer/password_change.html.erb
app/views/devise/mailer/reset_password_instructions.html.erb
app/views/devise/mailer/invitation_instructions.html.erb
app/views/devise/mailer/confirmation_instructions.html.erb
```

**Comment by ivan-kocienski-gfsc** [comment]
Yeah i was hoping that i didn't have to write tests for this as the whole thing is complicated by devise.

**Review by aaaaargZombies — APPROVED** [review]
yes, the old turn it on and off again fixed that problem. sorry for the misdirection

LGTM!

**Pull request merged**